backend/cmd/kubeconfig.go

```diff
@@ -30,7 +32,28 @@ func GetContextsFromKubeConfigFile(kubeConfigPath string) ([]Context, error) {
 		}
 
 		authInfo := config.AuthInfos[value.AuthInfo]
-		cluster := Cluster{key, clusterConfig.Server, clusterConfig}
+		authProvider := authInfo.AuthProvider
+
+		var authConfig AuthConfig
+
+		if authProvider != nil {
+			authType := authProvider.Name
+			clientID := authProvider.Config["client-id"]
+			clientSecret := authProvider.Config["client-secret"]
+			idpIssuerURL := authProvider.Config["idp-issuer-url"]
+
+			authConfig.ClientID = clientID
+			authConfig.ClientSecret = clientSecret
+			authConfig.IdpIssuerURL = idpIssuerURL
+			authConfig.Type = authType
+		} else {
+			authConfig.Type = "token"
```

Comment:
This is not correct, as you see in the minikube case, where it may have a client certificate.